packages/lib/product/service.ts

```diff
@@ -148,24 +148,32 @@ export const updateProduct = async (
   }
 };
 
-export const getProduct = async (productId: string): Promise<TProduct | null> => {
-  let productPrisma;
-  try {
-    productPrisma = await prisma.product.findUnique({
-      where: {
-        id: productId,
-      },
-      select: selectProduct,
-    });
+export const getProduct = async (productId: string): Promise<TProduct | null> =>
+  unstable_cache(
+    async () => {
+      let productPrisma;
+      try {
+        productPrisma = await prisma.product.findUnique({
+          where: {
+            id: productId,
+          },
+          select: selectProduct,
+        });
 
-    return productPrisma;
-  } catch (error) {
-    if (error instanceof Prisma.PrismaClientKnownRequestError) {
-      throw new DatabaseError(error.message);
+        return productPrisma;
+      } catch (error) {
+        if (error instanceof Prisma.PrismaClientKnownRequestError) {
+          throw new DatabaseError(error.message);
+        }
+        throw error;
+      }
+    },
+    [`getProduct-${productId}`],
+    {
+      tags: [productCache.tag.byId(productId)],
+      revalidate: SERVICES_REVALIDATION_INTERVAL,
     }
-    throw error;
-  }
-};
+  )();
 
 export const deleteProduct = async (productId: string): Promise<TProduct> => {
```

Comment:
that's a tricky one. As this removes the environments and everything in it:
- surveys
- responses
- tags
- tagOnResponses
- actions
- integrations
- apiKeys
- people
- responseNotes

No idea how to solve this 😅 But at least we have cache revalidation period, so they are deleted later.
So for now don't do further revalidations.